pipeline/transformer/transformer_common/src/main/scala/weco/pipeline/transformer/TransformerMain.scala

```diff
@@ -0,0 +1,70 @@
+package weco.pipeline.transformer
+
+import com.amazonaws.services.s3.AmazonS3
+import com.sksamuel.elastic4s.Index
+import io.circe.Decoder
+import weco.catalogue.internal_model.index.WorksIndexConfig
+import weco.catalogue.internal_model.work.Work
+import weco.catalogue.internal_model.work.WorkState.Source
+import weco.catalogue.internal_model.Implicits._
+import weco.catalogue.source_model.SourcePayload
+import weco.elasticsearch.typesafe.ElasticBuilder
+import weco.messaging.typesafe.SNSBuilder
+import weco.pipeline_storage.elastic.{ElasticIndexer, ElasticSourceRetriever}
+import weco.pipeline_storage.typesafe.PipelineStorageStreamBuilder
+import weco.storage.typesafe.S3Builder
+import weco.typesafe.WellcomeTypesafeApp
+import weco.typesafe.config.builders.AkkaBuilder
+import weco.typesafe.config.builders.EnrichConfig._
+
+import scala.concurrent.ExecutionContext
+
+trait TransformerMain[Payload <: SourcePayload, SourceData]
```

Comment:
ditto as above. Is there a reson to prefer a trait over a class with injected parameters? Reason for me to prefer a class in this case where we don't override the behavipour is that if it's a final class we can test the class once and trust that its behaviour cannot change where it used. 
In the case of traits it can be changed so tests on the trait are not enough imo, you need to test each implementation too with a pattern like the test cases. 
Also, I think not having to follow overrides makes for much readable code, and ingle test on a final class are also much more redable than TestCases.